syncer/syncer.go

```diff
@@ -1155,6 +1244,38 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 		utils.OsExit(1)
 	})
 
+	updateTSOffset := func() error {
+		t1 := time.Now()
+		ts, tsErr := s.fromDB.getServerUnixTS(ctx)
+		rtt := time.Since(t1).Seconds()
+		if tsErr == nil {
+			s.tsOffset.Store(time.Now().Unix() - ts - int64(rtt/2))
+		}
+		return tsErr
+	}
+	// before sync run, we get the tsoffset from upstream first
+	if utErr := updateTSOffset(); err != nil {
+		return utErr
+	}
+	// start background task to get/update current ts offset between dm and upstream
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+		// temporarily hard code there. if this metrics works well add this to config file.
+		updateTicker := time.NewTicker(time.Second * 10)
```

Comment:
maybe 10 minutes? I think offset won't change frequently, wait for other comment

Reply:
10 minutes is good enough